frontend/src/App.tsx

```diff
@@ -1,18 +1,41 @@
-import React from 'react';
+import React, { useEffect, useState } from "react";
+import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
+
 import "tabler-ui/dist/assets/css/dashboard.css";
-import './App.css';
-import Navbar from "./Navbar";
-import { BrowserRouter as Router, Route } from 'react-router-dom';
+import { AuthService } from "./service/authService";
+import { Login } from "./pages/login";
+import { Homepage } from "./pages/homepage";
+import { PrivateRoute } from "./utils/privateRoute";
+import { PageNotFoundError } from "./pages/errorPage";
+
+export const authService = new AuthService();
 
 const App: React.FC = () => {
-    return (<Router>
-        <div className="App">
-            <Route path="/">
-            <Navbar/>
-            </Route>
-        </div>
-        </Router>
-    );
+    // Check the authentication then renders the page
+    let [checkedAuth, setCheckedAuth] = useState(false);
+    useEffect(() => {
+        authService.checkAuth().then(() => {
+            setCheckedAuth(true);
+        });
```

Comment:
This effect is called at every render, consider specifying a condition for rendering

Reply:
![image](https://user-images.githubusercontent.com/4300678/68559888-81edc100-03f3-11ea-991f-ea6fbb0d12f7.png)

Reply:
Try to add a `console.log("bip")` in the function and to navigate to different pages, it'll be called only once, because the app isn't re-rendered

Reply:
OK, fine, au temps pour moi